sklearn/utils/_pprint.py

```diff
@@ -89,18 +89,27 @@ def _changed_params(estimator):
     estimator with non-default values."""
 
     params = estimator.get_params(deep=False)
-    filtered_params = {}
     init_func = getattr(estimator.__init__, 'deprecated_original',
                         estimator.__init__)
-    init_params = signature(init_func).parameters
+    init_params = inspect.signature(init_func).parameters
     init_params = {name: param.default for name, param in init_params.items()}
 
-    for k, v in params.items():
-        if (k not in init_params or (  # happens if k is part of a **kwargs
-                repr(v) != repr(init_params[k]) and
-                not (is_scalar_nan(init_params[k]) and is_scalar_nan(v)))):
-            filtered_params[k] = v
-    return filtered_params
+    def has_changed(k, v):
+        if k not in init_params:  # happens if k is part of a **kwargs
+            return True
+        if init_params[k] == inspect._empty:  # k has no default value
+            return True
+        # try to avoid calling repr on nested estimators
+        if (isinstance(v, BaseEstimator) and
```

Comment:
you could check if *either* v or init_params[k] is an estimator, in fact

Reply:
Is there code using estimators as default value ? I have never seen any and I doubt it would be nested estimators but I can add the check if you think it's useful.

Reply:
I have seen places (maybe not in sklearn itself) where the default value is an estimator, but then `v` in those cases is also an estimator. And since this is only a minor performance issue, I'm fine as is.